### DISCUSSION TITLE
How does MA determine how to classify local MP3's under Artists and Albums?

### DISCUSSION THREAD

**Discussion by FrankJaco** [music-assistant/support#538] [Q&A]
I have a 20,000 MP3 file collection currently on a Plex server. I am considering moving it to MA/HA as I only use music files, not any of plex's other features. I know from my plex experience the file naming conventions as well as the various MP3 tags are important to get right otherwise music can be mis-classified or artwork will not appear properly. Thus before moving in my large collection into MA, I wanted to do a few tests....

My first test was my collection of "3 Doors Down" which consists of music from 3 albums. I have then in a parent folder "3 doors down" and each of the 3 albums are inside named as the album names.  Plex displays them correctly, I used Picard/MusicBrains and MP3Tag to ensure everything is right. Yet when moving them into the media folder on HA/MA, it correctly identifies all the songs with artwork and correctly breaks it into 3 albums, but oddly it breaks it into two artists...  "3 Doors Down" and "Various Artists". I have checked the tags over and over and cannot figure out why this occurs.

So my bottom-line request, is if some can explain the actual logic MA uses, I would be grateful!

FJ

**Comment by OzGav** [comment]
Have a look at your tags for ALBUM ARTIST. I had a warning in my log from MA when one of my albums did not have this entry and it stated that it was going to use “Various Artists” as a default.

**Reply by FrankJaco** [reply]
Thanks for the reply, but I wish it was that easy. That is one of the first things I checked.   I went as far as clearing out all of the tags and saving it. Then used Picard-MusicBrains to once again to bring the correct tags in,   Still no luck. One thing that is truly puzzling is that when viewing all of the trasks in the track view, it actually correctly identifies the Artist! It is only in the artist folder that it is wrong.

BTW:  I checked the logs... And I do have an error which I am guessing is similar to what you experienced and may be pointing to the problem:

**Logger: music_assistant.metadata**
Source: /usr/local/lib/python3.9/site-packages/music_assistant/controllers/metadata/__init__.py:161
First occurred: June 27, 2022 at 8:55:18 AM (2 occurrences)
Last logged: June 27, 2022 at 6:04:55 PM

**Unable to get musicbrainz ID for artist Various Artists !**


FJ

**Comment by OzGav** [comment]
I'm no expert but that error seems to indicate that the ARTIST is " Various Artists". The ALBUM ARTIST is a different tag. I think you are probably aware of that but in my case the warning was definitely referring to ALBUM ARTIST

Edit. Also it says it occurred two times so it appears only two tracks have a problem?

**Reply by FrankJaco** [reply]
![Snap1](https://user-images.githubusercontent.com/79318542/176171698-535a9732-ac5b-49ff-bbbd-721e9e0017b1.jpg)
I agree that is exactly what it seems to be, but it unfortunately it is not.  See the attached image from MP3tag....   Both ARTIST and ALBUM ARTIST are set correctly. 
For now, I am putting this on hold.   I am in the process of moving my HA instance onto a M2. SSD in a generic micro-formfactor PC, and then will install another large SSD drive for the media etc.  Once things are back up and stable I will continue my quest.

Again thanks for your thoughts/ideas.

FJ

**Reply by marcelveldt** [reply]
Looking at the above screenshot I see absolutely no reason why the album got attached to various artists. Strange!
It's also the first time I've seen/heard this. Some people forget to set the albumartist tag, get confronted with the Various Artist fallback and correct their tags after which it is fine.
 
Did you check all files within the album folder ?

**Comment by marcelveldt** [comment]
Check here for all details how MA scans your files and what ID3 tags it reads, including additional metadata from the folders.

https://github.com/music-assistant/hass-music-assistant/discussions/543

**Comment by OzGav** [comment]
Relates to v1
